File: backend/internal/core/domain/deadline.go
package domain

import "time"

type DeadlineStatus string

const (
	StatusPending   DeadlineStatus = "PENDING"
	StatusOverdue   DeadlineStatus = "OVERDUE"
	StatusCompleted DeadlineStatus = "COMPLETED"
)

type Deadline struct {
	ID       string         `json:"id"`
	Title    string         `json:"title"`
	Category string         `json:"category"`
	DueDate  time.Time      `json:"dueDate"`
	Status   DeadlineStatus `json:"status"`
	Notes    string         `json:"notes,omitempty"`
}

File: frontend/src/types/deadline.ts
export type DeadlineStatus = "PENDING" | "COMPLETED" | "OVERDUE" | "UPCOMING" | string;

export type Deadline = {
  id: string;
  title: string;
  category: string;
  dueDate: string;
  notes: string;
  status: DeadlineStatus;
};

export type DeadlineCreateInput = {
  title: string;
  category: string;
  dueDate: string;
  notes: string;
};
